src/concerns/WithAuth.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { useLocation, Navigate } from 'react-router-dom';
+
+const withAuth = (Component) => {
+  function withAuthComponent(props) {
+    const path = useLocation().pathname;
+    const auth = localStorage.getItem('authentication');
+
+    if (!auth && !(['/login', '/signup'].includes(path))) {
```

Comment:
['/login', '/signup'].includes(path) could be on a variable so you don't need to call it twice
`const isAuthRoute = ['/login', '/signup'].includes(path)`